docs/config.md

```diff
@@ -0,0 +1,135 @@
+
+## config/config.go
+### Config
+- `metrics_address` — MetricsAddress (string). Directory to emit Prometheus metrics to. If `no_prometheus=true`, this will have no effect. Defaults to empty string, which means no metrics will be emitted.
+- `metrics_network` — MetricsNetwork (string). Chooses protocol to send metrics over (e.g. tcp, unix, etc). Defaults to empty string, which will send metrics via TCP.
+- `no_prometheus` — Defined [above](#configfsgofsconfig), cannot be redeclared
+- `debug_address` — DebugAddress (string). Address to send debug logs to. Defaults to empty string, which means no debugging metrics will be emitted.
+- `metadata_store` — MetadataStore (string). Determines how to store the metadata. Currently only a database is supported. Defaults to "db".
+
+# Categorized
+
+## config/fs.go
+
+### [http] RetryableHTTPClientConfig
+- `MaxRetries` — MaxRetries (int). Max amount of retries before giving up on a network request. Defaults to 0, which sets it to the value of defaultMaxRetries (8).
+- `MinWaitMsec` — MinWait (int). Minimum wait time, in milliseconds, between attempts for a network request. The actual wait time is governed by the BackoffStrategy, but the wait time will never be shorter than this duration. Defaults to 0, which sets it to the value of defaultMinWaitMsec (30).
+- `MaxWaitMsec` — MaxWait (int). Maximum wait time, in milliseconds, between attempts for a network request. The actual wait time is governed by the BackoffStrategy, but the wait time will never be longer than this duration. Defaults to 0, which sets it to the value of defaultMaxWaitMsec (30).
+- `DialTimeoutMsec` — DialTimeoutMsec (int). Maximum number of milliseconds that connection can take before a request attempt is timed out. Defaults to 0, which sets it to the value of defaultDialTimeoutMsec (3000).
+- `ResponseHeaderTimeoutMsec` — ResponseHeaderTimeoutMsec (int). Maximum duration waiting for response headers before a request attempt is timed out. This does not include reading the body. Defaults to 0, which sets it to the value of defaultResponseHeaderTimeoutMsec (3000).
+- `RequestTimeoutMsec` — RequestTimeoutMsec (int). Maximum duration before the entire request attempt is timed out. Defaults to 0, which sets it to the value of defaultRequestTimeoutMsec (30000).
+
+### [blob] BlobConfig
+- `valid_interval` — ValidInterval (int). Checks blob regularly at this interval in seconds. Defaults to 0, which sets it to the value of defaultValidIntervalSec (60).
+- `check_always` — CheckAlways (bool). Overrides `valid_interval` to be properly 0. Defaults to false.
+- `fetching_timeout_sec` — FetchTimeoutSec (int). Sets maximum amount of secodns to wait before failing to fetch. Defaults to 0, which sets it to the value of defaultFetchTimeoutSec (300).
+- `force_single_range_mode` — ForceSingleRnageMode (bool). Unused variable. Defaults to false.
+- `max_retries` — Overridden by MaxRetries in [[http]](#http)
```

Comment:
This is backwards right? This overrides the MaxRetries in http?

Reply:
Actually, is http even used? I recall looking into it and finding out that the http client is just legacy code from stargz and blob client is the actual client we use.

In that case, should we just remove the entire http section?

Reply:
HTTP is the global level HTTP configuration that applies to fetching spans or SOCI artifacts. The blob config only applies to fetching spans, so it overrides the retry configuration in the global HTTP config. We may want to remove the retry stuff from `BlobConfig`, I think it's mainly there for historical reasons. I don't see why we want blob specific HTTP knobs.